File: week6/problemset6/readability.py
from cs50 import get_string

def count_letters(text):
    letters = 0

    for i in range(len(text)):
        if (text[i].isalpha()):
            letters += 1

    return letters

def count_words(text):
    words = 1

    for i in range(len(text)):
        if (text[i] == ' '):
            words += 1

    return words

def count_sentences(text):
    sentences = 0

    for i in range(len(text)):
        if (text[i] == '.' or text[i] == '!' or text[i] == '?'):
            sentences += 1

    return sentences

def main():
    text = get_string("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    L = float(letters / words) * 100
    S = float(sentences / words) * 100
    index = 0.0588 * L - 0.296 * S - 15.8

    if (index < 1):
        print("Before Grade 1")
    elif (index >= 16):
        print("Grade 16+")
    else:
        print(f"Grade {index:.0f}")

if __name__ == "__main__":
    main()

File: week2/problemset2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

// declare variables - fixes undeclared function error
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

// main
int main(void)
{
    // prompt for text/sentences
    string text = get_string("Text: ");

    // do the counts : formula needs letters/sentences & sentences/word, need 3 variables
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // use formula
    // multiply by 100 not divide so because 'per 100 words'
    float L = ((float) letters / words) * 100;
    float S = ((float) sentences / words) * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;

    // print grade level: before grade 1, grade 16+, grade n
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", round(index));
    }
}

// count the letters
int count_letters(string text)
{
    int letters = 0;

    // count letters in text, add to letter for each alphanumeric character
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        // check if i is alphanumeric character
        if (isalpha(text[i]))
        {
            letters++;
        }
    }

    return letters;
}

// count the words
int count_words(string text)
{
    // initialize words, start at one since words wont have space
    int words = 1;

    // count words in text, add to variable after each space
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }
    return words;
}

// count number of sentences
int count_sentences(string text)
{
    int sentences = 0;

    // count sentences in text
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        // count ends of sentences where period/exclamation/question mark
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}
